src/components/Form/CreateProfileForm/ProfileData/ProfileData.tsx

```diff
@@ -59,20 +60,36 @@ const ProfileData = ({
   const agreement = useSelector((state: RootState) => state.agreement.value);
   const user = useSelector((state: RootState) => state.user.data);
   const [initialValues, setInitialValues] = useState<any>(defaultInitialValues);
+  const { triggerTooltip } = useTooltip();
   const [loading, setLoading] = useState(true);
   const [hasImage, setHasImage] = useState(false);
 
+  const throwErrorTooltip = () => {
+    triggerTooltip({
+      severity: Severity.DANGER,
+      text:
+        "Errore durante la creazione del profilo, controllare i dati e riprovare"
+    });
+  };
+
   const createProfile = (discount: any) => {
     if (agreement) {
-      void Api.Profile.createProfile(agreement.id, discount);
+      void Api.Profile.createProfile(agreement.id, discount)
+        .then(() => {
+          handleNext();
+        })
+        .catch(throwErrorTooltip);
     }
   };
 
   const updateProfile = (discount: any) => {
     if (agreement) {
-      void Api.Profile.updateProfile(agreement.id, discount).then(() =>
-        onUpdate()
-      );
+      void Api.Profile.updateProfile(agreement.id, discount)
+        .then(() => {
+          onUpdate();
+          handleNext();
+        })
+        .catch(throwErrorTooltip);
```

Comment:
Prefer use `taskEither` by wrapping Promise with `tryCatch` :)

Reply:
` await tryCatch(
      () => Api.Profile.updateProfile(agreement.id, discount),
      toError
    )
      .fold(throwErrorTooltip, () => {
        onUpdate();
        handleNext();
      })
      .run();`
      
      This never goes on Left when the API fail, I'm missing something?